website/addons/onedrive/serializer.py

```diff
@@ -0,0 +1,97 @@
+import logging
+
+from website.addons.base.serializer import OAuthAddonSerializer
+
+from website.util import api_url_for, web_url_for
+
+logger = logging.getLogger(__name__)
+
+
+class OneDriveSerializer(OAuthAddonSerializer):
+    def credentials_owner(self, user_settings=None):
+        return user_settings.owner or self.user_settings.owner
+
+    @property
+    def addon_short_name(self):
+        return 'onedrive'
+
+    @property
+    def user_is_owner(self):
```

Comment:
This should probably instead be the base implementation, the only difference between the two (`or len(user_accounts)`) was removed from the base version because it could cause bugs.